preact.js

```diff
@@ -0,0 +1,39 @@
+var Preact = require('preact')
+
+var Provider = (function (superclass) {
+  function Provider () {
+    superclass.apply(this, arguments)
+  }
+
+  if (superclass) Provider.__proto__ = superclass
+  Provider.prototype = Object.create(superclass && superclass.prototype)
+
+  Provider.prototype.constructor = Provider
+
+  Provider.prototype.getChildContext = function getChildContext () {
+    return { atom: this.props.atom }
+  }
+
+  Provider.prototype.render = function render (props) {
+    return props.children && props.children[0]
```

Comment:
in preact world, `props.children` is always an array, an empty one if there is no children